### ISSUE TITLE
학습 시나리오 0 테스트

### ISSUE THREAD

**Issue by TukBang** [TukBang/Pet-care-app#6]
Model: A1 ~ A6 (6-CLASS), EfficientNetV2_s
Preprocessing: 해상도 224x224 초과 이미지 제외, 환부 이외의 배경 포함 박스 이미지
batch_size: 128
optimizer = Adam
learning rate: 1e-4
criterion = CrossEntropyLoss

resize를 하지 않고 찍혀진 카메라의 구도에 맞게 학습한 경우 테스트 과정

**Comment by TukBang** [comment]
[Graph]
![graph](https://user-images.githubusercontent.com/121234327/231426275-09a54da2-2ce7-499c-9832-c45436905da8.png)

[AUC-ROC]
AUC per class: [0.899187940305332, 0.8913814505024766, 0.8795625755612606, 0.911454718264418, 0.906436881704622, 0.9313474910731065]
![image](https://user-images.githubusercontent.com/121234327/231426150-fa2bf29e-5497-4bc6-8dcd-55d21e486da9.png)

[classification report]

              precision    recall  f1-score   support

          A1       0.61      0.67      0.64       749
          A2       0.61      0.66      0.63       650
          A3       0.54      0.63      0.58       669
          A4       0.74      0.72      0.73      1020
          A5       0.80      0.65      0.72       992
          A6       0.81      0.70      0.75       313

    accuracy                           0.67      4393

**Comment by TukBang** [comment]
Scenario 0에서는 관심 영역이 큰 이미지를 제외한 2/3의 데이터 양에도 불구하고 균형 있는 정확도를 보여 주었고,
Scenario 2의 결과처럼 특징을 분할 해서 학습한다면 보다 잘 학습 할 것으로 판단

현재 데이터 셋에서는 환부가 잘 드러나도록 하게 끔 털을 최대한 안 보이게 하여 찍은 사진들이 대부분이고,
카메라의 구도에 따라서 같은 질환임에도 불구하고 관심 영역의 크기가 모두 다름에 따라서 Scale Problem이 발생함

이에 따라 Multi-Scale Problem에 관련된 Data Augumentation을 진행한 학습이 필요함

**Comment by TukBang** [comment]
Scenario 0에서는 다른 Scenario 1, 2, 3에서보다 AUC-ROC 커브의 수치가 높고,
관심 영역이 큰 이미지를 제외한 2/3의 데이터셋으로도 A1 ~ A6의 예측값들이 균형 있게 높음을 확인할 수 있었음

Scenario 3에서 알 수 있듯, 카메라 해상도 및 구도에 따른 클래스 별 예측 성능 차이가 생긴다는 점을 고려하여
관심 영역을 resize한 것에 더불어, 배경을 포함한 관심 영역을 가지는 것을 추가로 학습하는 Multi-Scale
Data Augumentation 방법을 적용하여 예측 성능 확인이 필요함

현재는 이미지를 resize할 시 크기가 작은 관심 영역은 이미지 픽셀이 깨지며, 이를 보완할 수 있을 것임
관심 영역이 분포에 따라서, 

[Scenario 4]
1. 작은 영역 (1분위수)는 box 이미지로부터 관심영역까지 3단계로 분할하여 증강 (x2)
2. 2, 3분위수는 box 이미지로부터 2단계까지 분할하여 증강 (x1)
3. 4분위 수는 증강하지 않음

[Scenario 5]
1. 관심영역이 너무 큰 이미지(4분위 수)를 제외한 3단계 증강

------------------------------------------------------------------------------

예측할 수 있는 문제점
1. 이미지 속에 있는 사람의 피부 등 배경이 관심 영역이 차지하는 비율보다 너무 커질 수 있음
2. 이미지의 증강이 너무 많아 메모리가 접근할 수 있는 리소스가 커질 수 있음

고려사항
1. 해당 방법은 box 정사각형의 이미지로 관심 영역이 매우 클 때 (예를 들어 1000 x 800), 이미지를 224 x 224로 resize할 시 zero-padding을 둘러 쌀 것인지, 1000x1000의 크기로 배경을 포함하여 줄일 것인지 고려
